Ignore arguments that don't match function literal or eta-expanded function in NoNeedForMonad

folone opened the pull request:
This is a quick and dirty fix for https://github.com/puffnfresh/wartremover/issues/142. The third case that @travisbrown mentioned is (I guess) something that extends `Function1`. This PR just ignores arguments like that (and all the other arguments) because the tree we're getting back looks like this `Ident(TermName("test"))`, and I'm not sure how to get my hands on `test.apply` tree here.

/cc @puffnfresh @vkostyukov

folone commented on core/src/test/scala/wartremover/warts/NoNeedForMonadTest.scala:
```diff
@@ -34,6 +34,14 @@ class NoNeedForMonadTest extends FunSuite {
+        def apply(i: Int) = Option(i + 2)
+      }
+
+      Option(1) flatMap test
```
The tree of the argument to `flatMap` here is `Ident(TermName("test"))`.

vkostyukov commented on core/src/test/scala/wartremover/warts/NoNeedForMonadTest.scala:
```diff
@@ -34,6 +34,14 @@ class NoNeedForMonadTest extends FunSuite {
     }
 
+    val extendsFunction = WartTestTraverser(NoNeedForMonad) {
+      object test extends Function1[Int, Option[Int]] {
```
Can we also add a `test` object that does not extend `Function1` but has method `apply`? Like this:

```scala
object test2 {
  def apply(i: Int) = Option(i + 4)
}

Option(2) flatMap test2
```

folone commented:
Bump :)

folone commented:
@travisbrown Yeah, there's an issue open for that: https://github.com/puffnfresh/wartremover/issues/130. I hope to get to it in the nearest future. Updated the versions and fixed the misspelling in 26fc1db.
